qemu/tests/acpi_index.py

```diff
@@ -0,0 +1,141 @@
+    acpi-index number, repeat step 7~9
+    11) Hot plug the nic with different slot and same acpi-index number, repeat
+    step 7~9
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def renew_ip_address(reboot_session, mac):
+        ifname = utils_net.get_linux_ifname(reboot_session, mac)
+        make_conf = "nmcli connection add type ethernet con-name %s ifname " \
+                    "%s autoconnect yes" % (ifname, ifname)
+        arp_clean = "arp -n|awk '/^[1-9]/{print \"arp -d \" $1}'|sh"
+        reboot_session.cmd_output_safe(make_conf)
+        reboot_session.cmd_output_safe("ip link set dev %s up" % ifname)
+        reboot_session.cmd_output_safe("dhclient -r", timeout=240)
+        reboot_session.cmd_output_safe("dhclient %s" % ifname, timeout=240)
+        reboot_session.cmd_output_safe(arp_clean)
+        return None
+
+    def verified_nic_name():
+        ifname = utils_net.get_linux_ifname(reboot_session,
+                                            vm.get_mac_address())
+        pattern = re.findall(r'\d+', ifname)[-1]
+        if pattern == "2":
+            logging.info("nic name match")
+        else:
+            test.fail("nic name doesn't match")
+
+    def ping_test():
+        host_ip = utils_net.get_host_ip_address(params)
+        status, output = utils_test.ping(host_ip, 10, timeout=30)
+        if status:
+            test.fail(
+                "%s ping %s unexpected, output %s" % (vm.name, host_ip, output))
+
+    def get_hotplug_nic_ip(vm, nic, reboot_session):
+        def __get_address():
+            try:
+                index = [_idx for _idx, _nic in enumerate(vm.virtnet)
+                         if _nic == nic][0]
+                return vm.wait_for_get_address(index, timeout=90)
+            except IndexError:
+                test.error("Nic '%s' not exists in VM '%s'"
+                           % (nic["nic_name"], vm.name))
+            except (virt_vm.VMIPAddressMissingError,
+                    virt_vm.VMAddressVerificationError):
+                renew_ip_address(reboot_session, nic["mac"])
+            return
+
+        nic_ip = utils_misc.wait_for(__get_address, timeout=360)
+        if nic_ip:
+            return nic_ip
+        cached_ip = vm.address_cache.get(nic["mac"])
+        arps = process.system_output("arp -aen").decode()
+        logging.debug("Can't get IP address:")
+        logging.debug("\tCached IP: %s", cached_ip)
+        logging.debug("\tARP table: %s", arps)
+        return None
+
+    login_timeout = int(params.get("login_timeout", 360))
+    repeat_times = int(params.get("repeat_times", 1))
+    logging.info("repeat_times: %s", repeat_times)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_serial_login(timeout=login_timeout)
+    remove_args = params["remove_args"]
+    session.cmd_output_safe(remove_args)
```

Comment:
It will automatically be removed if using `kernel_extra_params_remove`

Reply:
Hi @PaulYuuu I have made changes to the other questions you asked. The test passed. 
But for the parameter `kernel_extra_params_remove` , I hit two problems: 
1. For boot nic with option `acpi-index`. After deleting the parameters  `biosdevname=0 net.ifnames=0` and reboot guest, the ip cannot be regained.
2. For hotplug nic. Since nic is not assigned in the qemu command line, it also causes an error.
Therefore, I did not make any changes to this part. I put the error message in the bug. If you understand the reason, please let me know. Thanks in advance.